src/Features/Core/Portable/Completion/CompletionItem.cs

```diff
@@ -80,15 +98,26 @@ public sealed class CompletionItem : IComparable<CompletionItem>
             this.Rules = rules ?? CompletionItemRules.Default;
         }
 
-#pragma warning disable RS0027 // Public API with optional parameter(s) should have the most parameters amongst its public overloads.
         public static CompletionItem Create(
-#pragma warning restore RS0027 // Public API with optional parameter(s) should have the most parameters amongst its public overloads.
+            string displayText,
+            string filterText,
+            string sortText,
+            ImmutableDictionary<string, string> properties,
+            ImmutableArray<string> tags,
+            CompletionItemRules rules)
```

Comment:
should we care about consistency of order of parameters for the CompletionItem and CommonCompleitonItem?

Reply:
Right now they're completely inconsistent.  CompletionItem needs to work this way because it's a public API.  CommonCompletionItem is basically a grab bag of whatever we needed in whatever order we wanted over time.  We could reorder it if we wanted, but it didn't seem necessary.